instrumentation/hibernate/hibernate-6.0/src/main/java/com/solarwinds/opentelemetry/instrumentation/hibernate/v6_0/DrsaInstrumentation.java

```diff
@@ -0,0 +1,127 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.solarwinds.opentelemetry.instrumentation.hibernate.v6_0;
+
+import static io.opentelemetry.javaagent.bootstrap.Java8BytecodeBridge.currentContext;
+import static io.opentelemetry.javaagent.extension.matcher.AgentElementMatchers.hasClassesNamed;
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+import com.solarwinds.opentelemetry.instrumentation.hibernate.FunctionWrapper;
+import com.solarwinds.opentelemetry.instrumentation.hibernate.HibernateInstrumenter;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.javaagent.bootstrap.CallDepth;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import java.lang.reflect.Field;
+import java.sql.PreparedStatement;
+import java.util.function.Function;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess;
+
+public class DrsaInstrumentation implements TypeInstrumentation {
+
+  @Override
+  public ElementMatcher<ClassLoader> classLoaderOptimization() {
+    return hasClassesNamed("org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess");
+  }
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("org.hibernate.sql.results.jdbc.internal.DeferredResultSetAccess");
+  }
+
+  @Override
+  public void transform(TypeTransformer transformer) {
+    transformer.applyAdviceToMethod(
+        isMethod().and(named("executeQuery")),
+        DrsaInstrumentation.class.getName() + "$DeferredResultSetAccessAdvice");
+  }
+
+  @SuppressWarnings("unused")
+  public static class DeferredResultSetAccessAdvice {
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void startMethod(
+        @Advice.This DeferredResultSetAccess drsa,
+        @Advice.Local("swoCallDepth") CallDepth callDepth,
+        @Advice.Local("swoSqlContext") String swoSql,
+        @Advice.Local("swoContext") Context context,
+        @Advice.Local("swoScope") Scope scope) {
+      callDepth = CallDepth.forClass(DeferredResultSetAccess.class);
+      if (callDepth.getAndIncrement() > 0) {
+        return;
+      }
+
+      try {
+        // We're using reflection here because this seems to be the best place we can intercept the
+        // raw sql before the PreparedStatement is generated and be able to make the span a parent
+        // of the execution step.
```

Comment:
much appreciate the magic and the helpful comment above :) one question: the spans i've seen from testing are parametized and we don't collect actual argument values (which could contain sensitive info, good). but want to double check whether this would always be the case, e.g. is sanitization a concern for the `sql` value?

Reply:
It'll be a concern. However, I have removed adding of the sql during changes for v4 since it's not needed for the for feature to work.